ld-pruning-wf.wdl

```diff
@@ -0,0 +1,440 @@
+			f.write("ld_win_size ~{ld_win_size}\n")  # noquotes
+		if ~{maf_threshold} != 0.01:
+			f.write("maf_threshold ~{maf_threshold}\n")  # noquotes
+		if ~{missing_threshold} != 0.01:
+			f.write("missing_threshold ~{missing_threshold}\n")  # noquotes
+
+		if "~{def_sampleInc}" == "true":
+			f.write('sample_include_file "~{sample_include_file}"\n')
+
+		if "~{def_variantInc}" == "true":
+			f.write('variant_include_file "~{variant_include_file}"\n')
+
+		if "chr" in "~{gds}":
+			parts = os.path.splitext(os.path.basename("~{gds}"))[0].split("chr")
+			outfile_temp = "pruned_variants_chr" + parts[1] + ".RData"
+		else:
+			outfile_temp = "pruned_variants.RData"
+		if "~{out_prefix}" != "":
+			outfile_temp = "~{out_prefix}" + "_" + outfile_temp
+
+		f.write('out_file "' + outfile_temp + '"\n')
+
+		f.close()
+		CODE
+
+		echo "Calling R script ld_pruning.R"
+		Rscript /usr/local/analysis_pipeline/R/ld_pruning.R ld_pruning.config
+	}
+	
+	# Estimate disk size required
+	Int gds_size = ceil(size(gds, "GB"))
+	Int sample_size = if defined(sample_include_file) then ceil(size(sample_include_file, "GB")) else 0
+	Int varInclude_size = if defined(variant_include_file) then ceil(size(variant_include_file, "GB")) else 0
+	Int finalDiskSize = gds_size + sample_size + varInclude_size + addldisk
+
+	# Workaround for optional files
+	Boolean def_sampleInc = defined(sample_include_file)
+	Boolean def_variantInc = defined(variant_include_file)
+	
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master:2.10.0"
+		disks: "local-disk " + finalDiskSize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File ld_pruning_output = glob("*.RData")[0]
+		File config_file = "ld_pruning.config"
+	}
+
+}
+
+# [2] subset_gds -- subset a GDS file based on RData vector of variants
+task subset_gds {
+	input {
+		Pair[File, File] gds_n_varinc  # [gds, variant_include_file]
+		String? out_prefix
+		File? sample_include_file
+
+		# runtime attributes
+		Int addldisk = 1
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	command {
+		set -eux -o pipefail
+
+		python << CODE
+		import os
+
+		def py_rootPlusChr(py_filename):
+			'''
+			Similar to split_n_space in vcf-to-gds but not equivalent
+			The CWL uses nameroot for this, but I think the WDL needs the full path
+			Ex: "inputs/test_data_chrX.vcf.gz" returns ["inputs/test_data_", "X"]
+			'''
+			py_split = py_filename.split("chr")
+			if unicode(str(py_split[1][1])).isnumeric():
+				# chr10 and above
+				py_thisNamerootSplit = py_split[0]
+				py_thisChr = py_split[1][0:2]
+			else:
+				# chr9 and below + chrX
+				py_thisNamerootSplit = py_split[0]
+				py_thisChr = py_split[1][0:1]
+			return [py_thisNamerootSplit, py_thisChr]
+
+		gds = "~{gds_n_varinc.left}"
+		variant_include_file = "~{gds_n_varinc.right}"
+
+		f = open("subset_gds.config", "a")
+		f.write("gds_file " + gds + "\n")
+		f.write("variant_include_file " + variant_include_file + "\n")
```

Comment:
The CWL has this as optional
```
        	if (inputs.variant_include_file)
        	{
        	    config += 'variant_include_file "' + inputs.variant_include_file.path + "\"\n"
        	}
``` 
We need it to be optional as well.

Reply:
It's a hard requirement in the CWL, I'm not sure why Stephanie wrote it and the GDS as an if. https://github.com/UW-GAC/analysis_pipeline_cwl/blob/c68ac3f3c8b07512d0fcaffd03ed2d168e294993/relatedness/tools/subset_gds.cwl#L72